app/src/main/kotlin/compiler/ast/Expression.kt

```diff
@@ -23,6 +23,17 @@ sealed class Expression : AstNode {
         override val location: LocationRange? = null,
     ) : Expression()
 
+    data class ArrayElement(
+        val name: String,
```

Comment:
Shouldn't we have an `Expression` here (since we can return an array from a function or have array of an array).
(also, we have `Expression` in `ArrayLength`)